plugins/single_plugins/alpha.cpp

```diff
@@ -99,6 +77,47 @@ class wayfire_alpha : public wayfire_plugin_t
         }
     }
 
+    axis_callback axis_cb = [=] (wlr_event_pointer_axis* ev)
+    {
+        if (!output->activate_plugin(grab_interface))
+            return;
+
+        output->deactivate_plugin(grab_interface);
+
+        auto focus = core->get_cursor_focus();
+
+        if (!focus)
+            return;
+
+        auto view = core->find_view(focus->get_main_surface());
+        auto layer = output->workspace->get_view_layer(view);
+
+        if (layer == WF_LAYER_BACKGROUND)
+            return;
+
+        if (ev->orientation == WLR_AXIS_ORIENTATION_VERTICAL)
+            update_alpha(view, ev->delta);
+    };
+
+    wf_option_callback min_value_changed = [=] ()
+    {
+        output->workspace->for_each_view([=] (wayfire_view view)
+        {
+            wf_2D_view *transformer;
+
+            if (!view->get_transformer("alpha"))
+                return;
+
+            transformer = dynamic_cast<wf_2D_view*> (view->get_transformer("alpha").get());
+
+            if (transformer->alpha < min_value->as_double())
+                transformer->alpha = min_value->as_double();
+
+            view->damage();
```

Comment:
theoretically no need to damage if the alpha wasn't changed, could be moved inside the `if` on line 113

Reply:
True, fixed.